### PR TITLE
feat: Test Analytics Metrics buttons and small query linkages

### PR DESCRIPTION
**Pull request by ajay-sentry**
# Description

This PR aims to tackle the parameter filter for the TestResults query by converting some of the metrics in the metrics section to buttons, as well as gluing together totalFlakyTestCount, totalSlowestTests and totalSlowestTestsPercentChange from API to Gazebo

There are also a few related UT updates for these changes

# Screenshots

https://github.com/user-attachments/assets/002c5adf-7846-4ba7-85e1-c14a0a992aa3

Screenshot 2024-10-16 at 1 56 53 PM

# Link to Sample Entry

# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ const TestResultSchema = z.object({
   flakeRate: z.number().nullable(),
   avgDuration: z.number().nullable(),
   totalFailCount: z.number(),
+  totalFlakyFailCount: z.number(),
```
Comment (RulaKhaled) [inline src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx:26]: Love that we already have this

File: src/pages/RepoPage/FailedTestsTab/FailedTestsPage/MetricsSection/MetricsSection.tsx [hunk 2]
```diff
@@ -1,12 +1,16 @@
-import { useParams } from 'react-router-dom'
+import cs from 'classnames'
```
Comment (RulaKhaled) [inline src/pages/RepoPage/FailedTestsTab/FailedTestsPage/MetricsSection/MetricsSection.tsx:1]: We currently use:

`import { cn } from 'shared/utils/cn'
`
A simple in-place replacement should work

File: src/pages/RepoPage/FailedTestsTab/FailedTestsPage/FailedTestsTable/FailedTestsTable.tsx [hunk 3]
```diff
@@ -222,40 +222,45 @@ const FailedTestsTable = () => {
...
   const tableData = useMemo(() => {
+    if (!testData?.testResults) return []
+
     return (
-      testData?.testResults.map((result) => {
+      testData.testResults.map((result) => {
         const value = (result.flakeRate ?? 0) * 100
         const isFlakeInt = Number.isInteger(value)
 
+        const flakeRateContent = (
```
Comment (RulaKhaled) [inline src/pages/RepoPage/FailedTestsTab/FailedTestsPage/FailedTestsTable/FailedTestsTable.tsx:232]: A this returns JSX. Can you capitalize the first letter so it's FlakeRateContent?

File: src/pages/RepoPage/FailedTestsTab/FailedTestsPage/MetricsSection/MetricsSection.tsx [hunk 4]
```diff
@@ -192,7 +239,16 @@ const TotalFailuresCard = ({
         </MetricCard.Title>
       </MetricCard.Header>
       <MetricCard.Content>
-        {totalFails}
+        <button
+          className={cs('text-ds-blue-default hover:underline', {
+            'font-semibold': isSelected,
+          })}
+          onClick={() => {
+            updateParams({ parameter: 'FAILED_TESTS' })
```
Comment (RulaKhaled) [inline src/pages/RepoPage/FailedTestsTab/FailedTestsPage/MetricsSection/MetricsSection.tsx:247]: Pulling these out to an enum could be helpful

Comment (ajay-sentry) [reply src/pages/RepoPage/FailedTestsTab/FailedTestsPage/MetricsSection/MetricsSection.tsx:247]: I thought about it, I think the team has mixed feelings about enums and we haven't really come to a conclusion for when to use vs. not. I'll leave for now but will bring up in our next team meeting and update before launch if needed :)

**Pull request merged**